config/config.go

```diff
@@ -37,26 +51,31 @@ func parseTerragruntConfigFile(configPath string) (*TerragruntConfig, error) {
 }
 
 // Parse the Terragrunt config contained in the given string
-func parseTerragruntConfig(config string) (*TerragruntConfig, error) {
-	terragruntConfig := &TerragruntConfig{}
-
-	if err := hcl.Decode(terragruntConfig, config); err != nil {
+func parseConfigString(configSrc string) (*Config, error) {
+	f := &fileConfig{}
+	if err := hcl.Decode(f, configSrc); err != nil {
 		return nil, errors.WithStackTrace(err)
 	}
 
-	if terragruntConfig.DynamoDbLock != nil {
-		terragruntConfig.DynamoDbLock.FillDefaults()
-		if err := terragruntConfig.DynamoDbLock.Validate(); err != nil {
-			return nil, err
+	c := &Config{}
+
+	if lconf := f.Lock; lconf != nil {
```

Comment:
`lconf` made me wonder what a `left config` could be. `lockConfig` would be a clearer name.